bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java

```diff
@@ -603,7 +604,7 @@ private void replay(Journal journal, JournalScanner scanner) throws IOException
             journalId >= markedLog.getLogFileId());
         // last log mark may be missed due to no sync up before
         // validate filtered log ids only when we have markedLogId
-        if (markedLog.getLogFileId() > 0) {
+        if (markedLog.getLogFileId() > 0 && markedLog.getLogFileId() != Long.MAX_VALUE) {
```

Comment:
I wonder if it would be better to have a constant for Long.MAX_VALUE

Reply:
Change should also be made at `Journal.readLog`, which curMark should be set to the min mark from ledgers dir.